### PR TITLE
Added scaffolding for Schedule reducer

### PR DESCRIPTION
**Pull request by smarcet**
* added reload actions
* refactored reducers

ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2850915&fileview=grid

Signed-off-by: smarcet <smarcet@gmail.com>

### CODE REVIEW

**Comment by smarcet** [comment]
ie summit-schedule#event=24902&view=list

File: src/actions/schedule-actions.js [hunk 1]
```diff
@@ -7,30 +7,77 @@ export const UPDATE_FILTER      = 'UPDATE_FILTER';
...
+        if (eventEl) {
+            // avoid scroll cache
+            window.history.scrollRestoration = "manual";
+
+            // remove event from fragment
+            fragmentParser.deleteParam("event");
+
+            // scroll to event
+            setTimeout(() => {
+                eventEl.scrollIntoView({ behavior: "smooth", block: "center" });
```
Comment (gcutrini) [inline src/actions/schedule-actions.js:50]: Smooth scrolling still needs polyfilling in Safari.
Luckily was recently introduced in codebase [here](https://github.com/OpenStackweb/osf-website/pull/225/files#diff-ad9ea87f8488901c3102d09f1e42c10d60c96aecbce6cab0edccfc1b4faf5fccR9).

File: src/actions/schedule-actions.js [hunk 2]
```diff
@@ -80,4 +127,12 @@ export const getShareLink = (filters, view) => {
 
 export const callAction = (key, action, payload) => (dispatch) => {
     return dispatch(createAction(action)({...payload, key}));
+};
+
+export const reloadScheduleData = () => (dispatch, getState) => {
+    const {userState, loggedUserState} = getState();
```
Comment (gcutrini) [inline src/actions/schedule-actions.js:133]:
```suggestion
    const { userState: { userProfile }, loggedUserState: { isLoggedUser } } = getState();
```

File: src/actions/schedule-actions.js [hunk 3]
```diff
@@ -80,4 +127,12 @@ export const getShareLink = (filters, view) => {
...
+    const {isLoggedUser} = loggedUserState;
+    const {userProfile} = userState;
+    dispatch(createAction(RELOAD_SCHED_DATA)({isLoggedUser, userProfile }));
```
Comment (gcutrini) [inline src/actions/schedule-actions.js:136]: Actually, is isLoggedUser payload used at all?

File: src/actions/schedule-actions.js [hunk 4]
```diff
@@ -80,4 +127,12 @@ export const getShareLink = (filters, view) => {
...
+    dispatch(createAction(RELOAD_USER_PROFILE)({isLoggedUser, userProfile }));
```
Comment (gcutrini) [inline src/actions/schedule-actions.js:137]: Is this payload being used?

File: src/actions/schedule-actions.js [hunk 5]
```diff
@@ -7,30 +7,77 @@ export const UPDATE_FILTER      = 'UPDATE_FILTER';
...
 export const updateFilter = (key, filter, action = UPDATE_FILTER) => (dispatch) => {
     dispatch(createAction(action)({...filter, key}))
 };
 
-export const updateFiltersFromHash = (key, filters, view, actionCallback = UPDATE_FILTERS) => (dispatch) => {
-    const qsFilters = fragmentParser.getParams();
 
-    // clear hash
-    fragmentParser.clearParams();
+export const deepLinkToEvent = () => {
```
Comment (gcutrini) [inline src/actions/schedule-actions.js:31]: Should this be scoped as an action? Since it only resolves the presentational aspect.

**Review by gcutrini — COMMENTED** [review]
@smarcet please review comments, thanks.

File: src/actions/schedule-actions.js (hunk 1 above)
Comment (smarcet) [reply src/actions/schedule-actions.js:50]: ack

File: src/actions/schedule-actions.js (hunk 3 above)
Comment (smarcet) [reply src/actions/schedule-actions.js:136]: yes i do think so

File: src/actions/schedule-actions.js (hunk 4 above)
Comment (smarcet) [reply src/actions/schedule-actions.js:137]: yes i do think so

**Review by gcutrini — APPROVED** [review]
Looks good to me.

**Pull request merged**